templates/deploy-openstack.tpl

```diff
@@ -42,19 +42,72 @@ set +x
 export KAYOBE_VAULT_PASSWORD=$(cat ~/vault.password)
 set -x
 
+# Configure hosts
 kayobe control host bootstrap
 kayobe seed host configure
 kayobe overcloud host configure
 %{ if deploy_wazuh }kayobe infra vm host configure%{ endif }
 
+# Deploy Ceph
 kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/cephadm-deploy.yml
 sleep 30
 kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/cephadm.yml
 kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/cephadm-gather-keys.yml
 
+pip install -r $${config_directories[kayobe]}/requirements.txt
+
+# Deploy hashicorp vault to the seed
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/vault-deploy-seed.yml
+ansible-vault encrypt --vault-password-file ~/vault.password $KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/vault/OS-TLS-INT.pem
+ansible-vault encrypt --vault-password-file ~/vault.password $KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/vault/seed-vault-keys.json
+ansible-vault encrypt --vault-password-file ~/vault.password $KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/vault/overcloud.key
+
+kayobe overcloud service deploy -kt haproxy
+
+# Deploy hashicorp vault to the controllers
+kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/vault-deploy-overcloud.yml
```

Comment:
I believe in the original design it was supposed to be possible to run this without haproxy running, to avoid this bootstrapping issue. Did you test that?

Reply:
I did, and it didn't work. It fails with `role not found or permission denied` on controller 1 during the vault generate internal playbook.

Reply:
Did you look into what was going wrong? One thing you need to be careful of is targeting the active vault node.

Reply:
I haven't dug into it. I'll take another look

Reply:
@bbezak might be interested in this

Reply:
Ok, that's a shame. Let's proceed like this, and track the bootstrapping issue separately.